lib/score_card.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+class ScoreCard
+  attr_reader :frames
+
+  def initialize
+    @frames = []
+    @score = 0
+  end
+
+  def play(user_input)
+    final_frame ? tenth_round(user_input) : roll(user_input)
+  end
+
+  def roll(user_input)
```

Comment:
If you want to be able to test this independently from the Frame class using a double, you could use dependency injection by adding an argument with default value, e.g. `frame = Frame.new`. This wouldn't affect the functionality of the code, but it would allow you 'feed' a double into the method in the tests. Don't think it's entirely necessary, but it's an option.

Reply:
Yes, good shout! I was thinking of doing something like that too, but didn't get around to it yet!